examples/expositional/frameworks/cortexchat/cortex_chat_quickstart.ipynb

```diff
@@ -0,0 +1,303 @@
+   "source": [
+    "# Cortex Chat + TruLens\n",
+    "\n",
+    "This quickstart assumes you already have a Cortex Search Service started, JWT token created and Cortex Chat Private Preview enabled for your account. If you need assistance getting started with Cortex Chat, please contact your Snowflake contact."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set JWT Token and Chat URL"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"SNOWFLAKE_JWT\"] = \"...\"\n",
+    "os.environ[\"SNOWFLAKE_CHAT_URL\"] = \".../api/v2/cortex/chat\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a Cortex Chat App\n",
+    "\n",
+    "The `CortexChat` class below can be configured with your URL and model selection.\n",
+    "\n",
+    "It contains two methods: `handle_cortex_chat_response`, and `chat`.\n",
+    "- `_handle_cortex_chat_response` serves to handle the streaming response, and expose the debugging information.\n",
+    "- `chat` is a user-facing method that allows you to input a `query` and receive a `response` and `citaiton`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import json\n",
+    "from trulens.apps.custom import instrument\n",
+    "\n",
+    "class CortexChat:\n",
+    "    def __init__(self, url: str, model: str = \"mistral-large\"):\n",
+    "        \"\"\"\n",
+    "        Initializes a new instance of the CortexChat class.\n",
+    "        Parameters:\n",
+    "        url (str): The URL of the chat service.\n",
+    "        model (str): The model to be used for chat. Defaults to \"mistral-large\".\n",
+    "        \"\"\"\n",
+    "\n",
+    "        \n",
+    "        self.url = url\n",
+    "        self.model = model\n",
+    "\n",
+    "    @instrument\n",
+    "    def _handle_cortex_chat_response(self, response: requests.Response) -> tuple[str, str, str]:\n",
+    "        \"\"\"\n",
+    "        Process the response from the Cortex Chat API.\n",
+    "        Args:\n",
+    "            response: The response object from the Cortex Chat API.\n",
+    "        Returns:\n",
+    "            A tuple containing the extracted text, citation, and debug information from the response.\n",
+    "        Raises:\n",
+    "            json.JSONDecodeError: If there is an error decoding the JSON data in the response.\n",
+    "        \"\"\"\n",
+    "\n",
+    "        text = \"\"\n",
+    "        citation = \"\"\n",
+    "        debug_info = \"\"\n",
+    "        previous_line = \"\"\n",
+    "        \n",
+    "        for line in response.iter_lines():\n",
+    "            if line:\n",
+    "                decoded_line = line.decode('utf-8')\n",
+    "                if decoded_line == \"event: done\":\n",
```

Comment:
can we add a fallback return or raise an error in case this condition is never met as feedbacks are dependent on this response value